### PR TITLE
Reformatting of api.py, addition of extra examples, and inclusion of additional documentation in the README.

### PR DESCRIPTION
**Pull request by williampeoch**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
 
-An ergonomic way to use Metaphor in python.
+[![pypi](https://img.shields.io/pypi/v/metaphor-python.svg)](https://pypi.python.org/pypi/metaphor-python)
+
+
+The Metaphor Python library provides convenient access to the Metaphor API from
+applications written in the Python language. It includes a pre-defined set of
+classes for API resources that initialize themselves dynamically from API
+responses which makes it compatible with a wide range of versions of the Metaphor
+API.
```
Comment (jeffzwang) [inline README.md:10]: Nice to define this properly, but would be nice for this to be shorter

File: README.md [hunk 2]
```diff
@@ -34,6 +53,8 @@ for result in response.results:
     print(result.title, result.url)
 ```
 
+You can learn more in our [basic search guide](https://docs.metaphor.systems/reference/search-2).
```
Comment (jeffzwang) [inline README.md:56]: Dang realized these URL slugs were bad. Fixed - making suggestions here.
```suggestion
You can learn more in our [basic search guide](https://docs.metaphor.systems/reference/basic-search).
```

File: README.md [hunk 3]
```diff
@@ -43,6 +64,8 @@ for result in response.results:
     print(result.title, result.url)
 ```
 
+You can learn more in our [basic finding similar links guide](https://docs.metaphor.systems/reference/search-copy).
```
Comment (jeffzwang) [inline README.md:67]:
```suggestion
You can learn more in our [basic finding similar links guide](https://docs.metaphor.systems/reference/basic-finding-similar-links).
```

File: README.md [hunk 4]
```diff
@@ -53,6 +76,8 @@ for content in response.contents:
     print(content.title, content.url)
 ```
 
+You can learn more in our [basic content retrieval guide](https://docs.metaphor.systems/reference/get-contents-of-webpage).
```
Comment (jeffzwang) [inline README.md:79]:
```suggestion
You can learn more in our [basic content retrieval guide](https://docs.metaphor.systems/reference/basic-content-retrieval).
```

File: examples/basic_content_retrieval.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from metaphor_python import Metaphor
+
+client = Metaphor("fa0fcc27-312c-4d5e-b06d-f101f9717e91")
```
Comment (jeffzwang) [inline examples/basic_content_retrieval.py:3]:
```suggestion
client = Metaphor("your_key")
```

File: examples/basic_find_similar.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+from metaphor_python import Metaphor
+
+client = Metaphor("fa0fcc27-312c-4d5e-b06d-f101f9717e91")
```
Comment (jeffzwang) [inline examples/basic_find_similar.py:3]:
```suggestion
client = Metaphor("your_key")
```

**Review by jeffzwang — COMMENTED** [review]
thanks for this! left some feedback that is mainly our fault actually, just code suggestions for replacing

**Comment by Feel-ix-343** [comment]
Please reopen if this is still relevant

**Pull request closed without merging**